notifications/src/main/kotlin/com/amazon/opendistroforelasticsearch/commons/notifications/model/NotificationConfig.kt

```diff
@@ -0,0 +1,164 @@
+import org.elasticsearch.common.io.stream.StreamOutput
+import org.elasticsearch.common.io.stream.Writeable
+import org.elasticsearch.common.xcontent.ToXContent
+import org.elasticsearch.common.xcontent.XContentBuilder
+import org.elasticsearch.common.xcontent.XContentParser
+import org.elasticsearch.common.xcontent.XContentParserUtils
+import java.util.EnumSet
+
+/**
+ * Data class representing Notification config.
+ */
+data class NotificationConfig(
+    val name: String,
+    val configType: ConfigType,
+    val features: EnumSet<Feature>,
+    val isEnabled: Boolean = true,
+    val slack: Slack? = null,
+    val chime: Chime? = null,
+    val webhook: Webhook? = null
+) : Writeable, ToXContent {
+
+    init {
+        require(!Strings.isNullOrEmpty(name)) { "name is null or empty" }
+        when (configType) {
+            ConfigType.Slack -> requireNotNull(slack)
+            ConfigType.Chime -> requireNotNull(chime)
+            ConfigType.Webhook -> requireNotNull(webhook)
+            ConfigType.None -> log.info("Some config field not recognized")
+        }
+    }
+
+    enum class ConfigType { None, Slack, Chime, Webhook }
+    enum class Feature { None, Alerting, IndexManagement, Reports }
+
+    companion object {
+        private val log by logger(NotificationConfig::class.java)
+        private const val NAME_TAG = "name"
+        private const val CONFIG_TYPE_TAG = "configType"
+        private const val FEATURES_TAG = "features"
+        private const val IS_ENABLED_TAG = "isEnabled"
+        private const val SLACK_TAG = "slack"
+        private const val CHIME_TAG = "chime"
+        private const val WEBHOOK_TAG = "webhook"
+
+        /**
+         * reader to create instance of class from writable.
+         */
+        val reader = Writeable.Reader { NotificationConfig(it) }
+
+        /**
+         * Creator used in REST communication.
+         * @param parser XContentParser to deserialize data from.
+         */
+        fun parse(parser: XContentParser): NotificationConfig {
+            var name: String? = null
+            var configType: ConfigType? = null
+            var features: EnumSet<Feature>? = null
+            var isEnabled = true
+            var slack: Slack? = null
+            var chime: Chime? = null
+            var webhook: Webhook? = null
+
+            XContentParserUtils.ensureExpectedToken(
+                XContentParser.Token.START_OBJECT,
+                parser.currentToken(),
+                parser::getTokenLocation
+            )
+            while (parser.nextToken() != XContentParser.Token.END_OBJECT) {
+                val fieldName = parser.currentName()
+                parser.nextToken()
+                when (fieldName) {
+                    NAME_TAG -> name = parser.text()
+                    CONFIG_TYPE_TAG -> configType = valueOf(parser.text(), ConfigType.None)
+                    FEATURES_TAG -> features = parser.enumSet(Feature.None)
```

Comment:
Why is this always None?

Reply:
because if new feature gets added and is not parable then it will fallback to None so that code safely ignores until upgraded